realm/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.test.AndroidTestCase;
+
+import io.realm.entities.Dog;
+import io.realm.entities.Owner;
+
+public class RealmListTest extends AndroidTestCase {
+
+    public static final int TEST_OBJECTS = 10;
```

Comment:
There are many dependencies to the magic number 10 in the below tests. That's ok if we never intend to change TEST_OBJECTS. But then at least make a comment about it here. Or make all the checks depend on TEST_OBJECTS (but it might not be worth while).